python/src/spark_rapids_ml/core.py

```diff
@@ -400,6 +401,57 @@ def _require_nccl_ucx(self) -> Tuple[bool, bool]:
         """
         return (True, False)
 
+    def _validate_parameters(self) -> None:
+        cls_name = self.__class__.__name__
+
+        pyspark_est: Optional[JavaParams] = None
+
+        # Is there a better way for the dynamic imports?
+        if "LogisticRegression" in cls_name:
+            from pyspark.ml.classification import LogisticRegression
+
+            pyspark_est = LogisticRegression()
+        elif "RandomForestClassifier" in cls_name:
+            from pyspark.ml.classification import RandomForestClassifier
+
+            pyspark_est = RandomForestClassifier()
+        elif "RandomForestRegressor" in cls_name:
+            from pyspark.ml.regression import RandomForestRegressor
+
+            pyspark_est = RandomForestRegressor()
+        elif "LinearRegression" in cls_name:
+            from pyspark.ml.regression import LinearRegression
+
+            pyspark_est = LinearRegression()
+        elif "PCA" in cls_name:
+            from pyspark.ml.feature import PCA
+
+            pyspark_est = PCA()
+        elif "KMeans" in cls_name:
+            from pyspark.ml.clustering import KMeans
+
+            pyspark_est = KMeans()
+        if pyspark_est is not None:
+            # Both pyspark and cuml may have a parameter with the same name,
```

Comment:
In this case, wouldn't the value stored in the Spark param be the unmapped value and not the cuml-specific value?   So seems it should be ok.   It's the other case that might be a problem (setting a cuml param value directly) - see other comment.